core.go

```diff
@@ -412,6 +418,49 @@ func (m *Mat) SetDoubleAt3(x, y, z int, val float64) {
 	C.Mat_SetDouble3(m.p, C.int(x), C.int(y), C.int(z), C.double(val))
 }
 
+// ToImage returns Go image.Image from Mat.
+func (m *Mat) ToImage() (image.Image, error) {
+	t := m.Type()
+	if t != int(MatTypeCV8U) && t != int(MatTypeCV8UC1) && t != int(MatTypeCV8UC3) && t != int(MatTypeCV8UC4) {
```

Comment:
Maybe we could create a loop instead and:

1. cast the constant only once
2. return the type in the error message (like "mat type is X, expected on of these: {Y, Z}")

Reply:
This also makes me think that the `Type()` function shoukd return `MatType`.

Reply:
Now `t` is casted to MatType and without MatTypeCV8U it looks better?